turbinia/workers/__init__.py

```diff
@@ -571,14 +574,26 @@ def execute(
 
     # Execute the job on the host system.
     else:
-      if shell:
-        proc = subprocess.Popen(
-            cmd, shell=True, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
-        proc.wait(timeout_limit)
-      else:
-        proc = subprocess.Popen(
-            cmd, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
-        proc.wait(timeout_limit)
+      try:
+        if shell:
+          proc = subprocess.Popen(
+              cmd, shell=True, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+          proc.wait(timeout_limit)
+        else:
+          proc = subprocess.Popen(
+              cmd, stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+          proc.wait(timeout_limit)
+      except subprocess.TimeoutExpired as exception:
+        # Log error and close result.
+        message = 'Execution of [{0!s}] failed due to job timeout of seconds has been reached.'.format(
+            cmd)
+        result.log(message)
+        if close:
```

Comment:
I wonder if we need to check if close is set or not.  If the exception raised here will bypass the rest of the `run()` method and get caught in `run_wrapper()` I think we don't need to.  The only reason not to close the result in the `execute()` method is if something else that gets run in the `run()` method after this is setting a custom status or adding more things to the result, so if we're bypassing all that then those things won't happen in this case.

Reply:
that's what the if statement is doing unless I'm missing something else?

Reply:
ah okay I understand, I went ahead and removed that check.